### ISSUE TITLE
UploadContainer And Image Metadata

### ISSUE THREAD

**Issue by dtcblyth** [elegantthemes/Divi-Beta#37]
Related thread: https://discord.com/channels/1041765492907589683/1157236208762638379

### Problem:
When using the `<UploadContainer />` component for images, there doesn’t appear to any built-in helper functions for getting the image’s metadata, such as: `titleText`, `altText`, `caption`, `description`, etc.

Currently we can only get these values using the RestAPI, but this requires an additional call to the server. Furthermore, because the resolved value for the `<UploadContainer />` is the image `url`, we are forced to use the `attachment_url_to_postid()` function to get the image post `id`.

### Solution:
Considering the Visual Builder is fetching the image anyway, would it not be possible to also get the images metadata at the same time? Instead of returning only the image URL, perhaps the `<UploadContainer />` could return an object which includes, the image `url`, `id`, and other metadata.

**Comment by eduard-un** [comment]
Hey @dtcblyth,

I think this should have been resolved with the release of the Custom Attribute feature. 

Do you think it would be possible to check if this is still an issue?

Let me know how it goes. Thanks 😄

**Comment by dtcblyth** [comment]
@eduard-un I don’t think this is related to, or solved by, the _Custom Attributes_ feature. This was a **feature request** to have the Divi API fetch not just an uploaded image’s `url`, but **all** of its meta data. This would simplify the process of displaying an image and using its meta data in the _Visual Builder_.

Currently, I fetch the meta data myself from the image `url` using `attachment_url_to_postid()`. However, this requires a second separate fetch request, which seems redundant considering that Divi is fetching the `url` anyway.

**Comment by eduard-un** [comment]
Hey @dtcblyth,

Thanks for the clarification. I'm going to forward this to our dev team and this will be tracked on our internal D5 Gir Repo. I'm keeping this one open, in case I need to share updates with you.

**Comment by dtcblyth** [comment]
@eduard-un Thanks for the update. I would say this is not an urgent issue. As previously mentioned, I have a working solution. It is more just a feature request / optimisation. Thanks.
